src/profile-logic/profile-data.js

```diff
@@ -569,6 +569,234 @@ export function filterThreadToPrefixCallNodePath(
+      const prefixDepth = prefix === null ? -1 : stackDepths[prefix];
+      const currentFuncOnPath = prefixCallNodePath[prefixDepth + 1];
+
+      let doMerge = false;
+      let stackDepth = prefixDepth;
+      let doesMatchCallNodePath;
+      if (doesPrefixMatch && stackDepth < depthAtCallNodePathLeaf) {
+        // This stack's prefixes were in our CallNodePath.
+        if (currentFuncOnPath === funcIndex) {
+          // This stack's function matches too!
+          doesMatchCallNodePath = true;
+          if (stackDepth + 1 === depthAtCallNodePathLeaf) {
+            // Holy cow, we found a match for our merge operation and can merge this stack.
+            doMerge = true;
+          } else {
+            // Since we found a match, increase the stack depth. This should match
+            // the depth of the implementation filtered stacks.
+            stackDepth++;
+          }
+        } else if (!funcMatchesImplementation(thread, funcIndex)) {
+          // This stack's function does not match the CallNodePath, however it's not part
+          // of the CallNodePath's implementation filter. Go ahead and keep it.
+          doesMatchCallNodePath = true;
+        } else {
+          // While all of the predecessors matched, this stack's function does not :(
+          doesMatchCallNodePath = false;
+        }
+      } else {
+        // This stack is not part of a matching branch of the tree.
+        doesMatchCallNodePath = false;
+      }
+      stackMatches[stackIndex] = doesMatchCallNodePath;
+      stackDepths[stackIndex] = stackDepth;
+
+      // Map the oldStackToNewStack, and only push on the stacks that aren't merged.
+      if (doMerge) {
+        const newStackPrefix = oldStackToNewStack.get(prefix);
+        oldStackToNewStack.set(
+          stackIndex,
+          newStackPrefix === undefined ? null : newStackPrefix
+        );
+      } else {
+        const newStackIndex = newStackTable.length++;
+        const newStackPrefix = oldStackToNewStack.get(prefix);
+        newStackTable.prefix[newStackIndex] =
+          newStackPrefix === undefined ? null : newStackPrefix;
+        newStackTable.frame[newStackIndex] = frameIndex;
+        oldStackToNewStack.set(stackIndex, newStackIndex);
+      }
+    }
+    const newSamples = Object.assign({}, samples, {
+      stack: samples.stack.map(oldStack => {
+        const newStack = oldStackToNewStack.get(oldStack);
+        if (newStack === undefined) {
+          throw new Error(
+            'Converting from the old stack to a new stack cannot be undefined'
+          );
+        }
+        return newStack;
+      }),
+    });
+    return Object.assign({}, thread, {
+      stackTable: newStackTable,
+      samples: newSamples,
+    });
+  });
+}
+
+/**
+ * Transform a thread to merge stacks that match the `postfixCallNodePath` into
+ * the caller. See `src/types/transforms.js` for more information about the
+ * "merge-call-node" transform.
+ */
```

Comment:
honnestly I think I would have just disabled the "merge to caller" item in inverted mode. Because we're not really merging to caller, right ? This is confusing at best... And is it even useful ?